enhancements/installer/ovirt-rhv-upi.md

```diff
@@ -0,0 +1,166 @@
+on oVirt using existing user-provisioned infrastructure such as DNS, DHCP, Load Balancers.
+
+Currently, users who want to deploy OpenShift on existing infrastructure in oVirt must
+adapt the [bare metal UPI instructions][baremetal-upi], which are not optimal for oVirt.
+
+Using the current installation program for deploying OpenShift 4 on oVirt does not work
+because it creates an opinionated set of _installer-provisioned infrastructure_ (IPI).
+Though convenient, this IPI does not meet the needs of users who want to use _existing_
+infrastructure.
+
+### Goals
+
+The main goal of the UPI is to grant users a higher degree of freedom compared to the
+more opinionated installer-provisioned installation.
+
+As a first step, we would create a provisioning installer codebase and a set of documented steps.
+These would enable users to deploy OpenShift on UPI in a way that is very similar to IPI but
+reusing and customizing pre-existing infrastructure elements.
+
+Following other UPI installer directives, this first step would include
+
+* Making oVirt-RHV UPI documentation available here:
+https://github.com/openshift/installer/blob/master/docs/user/ovirt/install_upi.md
+* Making Ansible playbooks for scripts for oVirt/RHV resource creation available here:
+https://github.com/openshift/installer/tree/master/upi/ovirt
+* Making a CI job executing the provisioning scripts to test the UPI installer
+
+### Non-Goals
+
+It is outside the scope of this enhancement to provide explanations about the installation
+of infrastructure elements that are considered as required and owned by the user (e.g. DNS,
+DHCP, Load Balancer...)
+
+## Proposal
+
+* Write Ansible playbooks that can automate as much as possible the creation of ovirt resources
+like the VMs that will be used as masters and worker nodes of the cluster.
+* Write documentation and Ansible scripts to help users configure pre-existing infrastructure
+elements, such as DNS, DHCP
+* Write the UPI documentation linking Ansible playbooks
+* Setup the CI job to have running test suite
+
+
+### Implementation Details/Notes/Constraints
+
+The implementation of the UPI workflow aims to reproduce the features of the IPI that enable
+users to deploy OpenShift on existing infrastructure.
+
+The user should be able to specify custom configurations that use the pre-existing
+infrastructure for the installation, such as the domain name, DNS, DHCP, Load Balancer, Web server.
+
+Given the maturity of the oVirt Ansible modules and administrators' familiarity with Ansible,
+we will provide a set of playbooks to automate the configuration and installation process as much
+as possible.
+
+Depending on the user environment, we will provide instructions and scripts that could help the user:
+
+* Add mandatory records (A, PTR, SRV) to the user-provided DNS.
+* Upload a specific RHCOS image.
+* Configure an httpd server for RHCOS customized parameter ignition.
```

Comment:
How does this work in IPI? I'm surprised by this need.

Reply:
This is not needed anymore, 'cause We can use custom_script also via ansible to pass ignition configs with the ansible ovirt_vm